src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -84,84 +55,138 @@ OTARequestorInterface * GetRequestorInstance()
+                     nodeIdString.data());
+        return CHIP_ERROR_INVALID_DESTINATION_NODE_ID;
+    }
 
-    CHIP_ERROR err = CHIP_NO_ERROR;
+    nodeId = chip::Encoding::BigEndian::Get64(buffer);
+    if (!IsOperationalNodeId(nodeId))
+    {
+        ChipLogError(SoftwareUpdate, "Image URI contains invalid node ID: 0x" ChipLogFormatX64, ChipLogValueX64(nodeId));
+        return CHIP_ERROR_INVALID_DESTINATION_NODE_ID;
+    }
 
-    // TODO: handle QueryImageResponse status types
-    if (response.status != EMBER_ZCL_OTA_QUERY_STATUS_UPDATE_AVAILABLE)
+    // Verify the separator between authority and path fields
+    CharSpan separator(bdxSeparator, strlen(bdxSeparator));
+    if (!uri.SubSpan(strlen(bdxPrefix) + kNodeIdHexStringMaxLen, strlen(bdxSeparator)).data_equal(separator))
     {
-        return;
+        return CHIP_ERROR_MISSING_URI_SEPARATOR;
     }
 
-    VerifyOrReturn(mBdxDownloader != nullptr, ChipLogError(SoftwareUpdate, "downloader is null"));
+    // Extract file designator from the path field
+    size_t fileDesignatorLength = uri.size() - (strlen(bdxPrefix) + kNodeIdHexStringMaxLen + strlen(bdxSeparator));
+    fileDesignator              = uri.SubSpan(uri.size() - fileDesignatorLength, fileDesignatorLength);
 
-    // TODO: allow caller to provide their own OTADownloader instance and set BDX parameters
+    return CHIP_NO_ERROR;
+}
+
+void OTARequestor::OnQueryImageResponse(void * context, const QueryImageResponse::DecodableType & response)
+{
+    LogQueryImageResponse(response);
 
-    TransferSession::TransferInitData initOptions;
-    initOptions.TransferCtlFlags = chip::bdx::TransferControlFlags::kReceiverDrive;
-    initOptions.MaxBlockSize     = 1024;
-    char testFileDes[9]          = { "test.txt" };
-    initOptions.FileDesLength    = static_cast<uint16_t>(strlen(testFileDes));
-    initOptions.FileDesignator   = reinterpret_cast<uint8_t *>(testFileDes);
+    VerifyOrReturn(context != nullptr, ChipLogError(SoftwareUpdate, "Received QueryImageResponse with invalid context"));
 
-    chip::OperationalDeviceProxy * operationalDeviceProxy = Server::GetInstance().GetOperationalDeviceProxy();
-    if (operationalDeviceProxy != nullptr)
+    OTARequestor * requestorCore = static_cast<OTARequestor *>(context);
+
+    switch (response.status)
     {
-        chip::Messaging::ExchangeManager * exchangeMgr = operationalDeviceProxy->GetExchangeManager();
-        chip::Optional<chip::SessionHandle> session    = operationalDeviceProxy->GetSecureSession();
-        if (exchangeMgr != nullptr && session.HasValue())
-        {
-            exchangeCtx = exchangeMgr->NewContext(session.Value(), &mBdxMessenger);
-        }
+    case EMBER_ZCL_OTA_QUERY_STATUS_UPDATE_AVAILABLE: {
+        VerifyOrReturn(response.imageURI.HasValue(), ChipLogError(SoftwareUpdate, "Update is available but no image URI present"));
```

Comment:
The mandatory logging with void return here prevents clients from understanding/recording the fact that an erroneous response was received. Clients should be able to account for this without humans reading logs.

Reply:
What do you propose here? `OnQueryImageResponse` is the callback after a QueryImage request was made. If there are failures here, the process of query stops. Are you suggesting requestorCore do something with this?

Reply:
You could add a method to OTARequestorDriver method that is used to report error conditions with an OtaRequestorError struct/class that encapsulates the necessary data for different errors (e.g. classic `TypeEnum errorType;` and a union of the different payloads underneath. This would allow the implementers to do the requisite tracking with their instrumentation and possibly decide to retry earlier for a query, for instance.

Reply:
This is beyond the purpose of this PR (only try to hook up CSM here). I'll file a TODO to add a way to report error status back to the OTARequestorDriver.

Reply:
Yes, please add a TODO, since there is additional error handling being done here. If we say the CSM is the only scope, but there is a lot more error-handling happening, then it gets confusing.